packages/suite/src/views/wallet/sign-verify/index.tsx

```diff
@@ -74,155 +76,190 @@ interface FormEvent {
-                                rows={4}
-                                maxRows={4}
-                                maxLength={255}
-                            />
-                        </Row>
-                        <Row>
-                            <Textarea
-                                topLabel={intl.formatMessage(messages.TR_SIGNATURE)}
-                                name="verifySignature"
-                                value={verifySignature}
-                                onChange={this.handleInputChange}
-                                rows={4}
-                                maxRows={4}
-                                maxLength={255}
-                            />
-                        </Row>
-                        <RowButtons>
-                            <StyledButton onClick={signVerifyActions.clearVerify}>
-                                <Translation id="TR_CLEAR" />
-                            </StyledButton>
-                            <StyledButton
-                                // isDisabled={!!verifyAddressError || !device.connected}
-                                isDisabled
-                                onClick={() => {
-                                    if (errors.length <= 0) {
-                                        signVerifyActions.verify(
-                                            verifyAddress,
-                                            verifyMessage,
-                                            verifySignature,
-                                        );
-                                    }
-                                }}
-                            >
-                                <Translation id="TR_VERIFY" />
-                            </StyledButton>
-                        </RowButtons>
-                    </Verify>
-                </Wrapper>
-            </WalletLayout>
-        );
-    }
-}
+    const handleInputChange = (event: FormEvent) => {
+        signVerifyActions.inputChange(event.target.name as InputNameType[0], event.target.value);
+    };
+
+    const getAddress = (): AddressToSign => {
+        if (
+            selectedAccount.account.networkType === 'bitcoin' &&
+            typeof selectedAccount.account.addresses?.unused[0]?.address === 'string' &&
+            typeof selectedAccount.account.addresses?.unused[0]?.address === 'string'
+        ) {
+            return {
+                address: selectedAccount.account.addresses.unused[0].address,
+                path: selectedAccount.account.addresses.unused[0].path,
+            };
+        }
+
+        if (selectedAccount.account.networkType === 'ethereum') {
+            return {
+                address: selectedAccount.account.descriptor,
+                path: selectedAccount.account.path,
+            };
+        }
+
+        return {
+            address: '',
+            path: '',
+        };
+    };
+
+    const verifyAddressError = getError('verifyAddress');
+
```

Comment:
consider to call `getAddress()` once here and use it in lower scope (right now you're calling same function multiple times

Reply:
actually, in fact, through the hook that you told me to use, I had to get some values also through function calls (getValues () ), but OK, I rewritten it